gems/aws-sdk-core/spec/aws/assume_role_web_identity_credentials_spec.rb

```diff
@@ -143,18 +143,32 @@ module Aws
       c = AssumeRoleWebIdentityCredentials.new(
         role_arn: 'arn',
         web_identity_token_file: token_file_path,
-      ) 
+      )
       expect(c).to be_set
       expect(c.credentials.access_key_id).to eq('akid')
       expect(c.credentials.secret_access_key).to eq('secret')
       expect(c.credentials.session_token).to eq('session')
       expect(c.expiration).to eq(in_one_hour)
     end
 
+    it 'refreshes asynchronously' do
```

Comment:
This test is funky, for some reason I'm seeing this error:

```
/Users/danieldao/.rbenv/versions/2.5.8/lib/ruby/gems/2.5.0/gems/rspec-mocks-3.10.3/lib/rspec/mocks/space.rb:51:in `raise_lifecycle_message': The use of doubles or partial doubles from rspec-mocks outside of the per-test lifecycle is not supported. (RSpec::Mocks::OutsideOfExampleError)
```

🤔

Reply:
I believe the double was being used in the Thread which was still running after the test finished - I fixed by updating the test to use `expect(Thread).to receive(:new).and_yield` which will run the thread's block synchronously.

Reply:
Oh interesting! [TIL](https://blog.petitviolet.net/post/2021-09-25/make-thread-synchronous-in-rspec) that `and_yield` forces the block to run sync.